src/daft-dsl/src/functions/utf8/rpad.rs

```diff
@@ -0,0 +1,53 @@
+use crate::ExprRef;
+use daft_core::{
+    datatypes::{DataType, Field},
+    schema::Schema,
+    series::Series,
+};
+
+use crate::functions::FunctionExpr;
+use common_error::{DaftError, DaftResult};
+
+use super::super::FunctionEvaluator;
+
+pub(super) struct RpadEvaluator {}
+
+impl FunctionEvaluator for RpadEvaluator {
+    fn fn_name(&self) -> &'static str {
+        "rpad"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema, _: &FunctionExpr) -> DaftResult<Field> {
+        match inputs {
+            [data, length, pad] => {
+                let data = data.to_field(schema)?;
+                let length = length.to_field(schema)?;
+                let pad = pad.to_field(schema)?;
+                if data.dtype == DataType::Utf8
+                    && length.dtype.is_integer()
+                    && pad.dtype == DataType::Utf8
+                {
+                    Ok(Field::new(data.name, DataType::Utf8))
+                } else {
+                    Err(DaftError::TypeError(format!(
+                    "Expects inputs to rpad to be utf8, integer and utf8, but received {}, {}, and {}", data.dtype, length.dtype, pad.dtype
+                )))
```

Comment:
Small nit: indent these lines
```suggestion
                        "Expects inputs to rpad to be utf8, integer and utf8, but received {}, {}, and {}", data.dtype, length.dtype, pad.dtype
                    )))
```